commons/all/src/main/java/org/infinispan/commons/configuration/attributes/AttributeSet.java

```diff
@@ -59,13 +60,14 @@ public AttributeSet(Class<?> klass, String name, AttributeSet attributeSet, Attr
    public AttributeSet(Class<?> klass, String name, AttributeSet attributeSet, AttributeDefinition<?>[] attributeDefinitions, RemovedAttribute[] removedAttributes) {
       this.klass = klass;
       this.name = name;
+      Map<String, Attribute<?>> attributes;
       if (attributeSet != null) {
-         this.attributes = new LinkedHashMap<>(attributeDefinitions.length + attributeSet.attributes.size());
```

Comment:
Are we sure we want to change this to be a HashMap? All of the methods including toString would probably be okay but the write method for the xml writing would change the ordering of the elements written which could be a maintenance pain. WDYT?

Reply:
Is this change why https://github.com/infinispan/infinispan/pull/11532/commits/45b1af0bfeef39f75bf7f8d15fc027be2e36d7c7 is needed?